1-Vanilla/vanillaTodoListCopy.js

```diff
@@ -0,0 +1,127 @@
+
+(function () {
+
+const createCheck = (todoId) => {
+    let checkDiv = document.createElement('input');
+    checkDiv.type = 'checkbox'
+    const checkDivClassName = "checked"
+    checkDiv.classList.add(checkDivClassName)
+    checkDiv.onclick = function(){handleCheckedClick(todoId)}
+    return checkDiv
+}
+
+const createButton = (type, todoId) => {
+    let button = document.createElement('button');
+    const buttonClassName = `${type}-button`
+    button.classList.add(buttonClassName)
+    button.appendChild(createIcon(type, todoId))
+    button.onclick = function(){handleButtonClick(type, todoId)}
+    return button
+} 
+const createIcon = (type, id)  => {
+    debugger;
+    let icon = document.createElement('i');
+    let iconClassName = ""
+    let iconId = ``
+    debugger
+    switch(type) {
+        case 'delete':
+            iconClassName = "fas fa-trash";
+            iconId = `icon-delete-${id}`
+            break;
+        case 'edit':
+            iconClassName = "fas fa-edit";
+            iconId = `icon-edit-${id}`
+            break;
+        case 'check':
+            iconClassName = "fa-check";
+            iconId = `icon-check-${id}`
+        default:
+            // none
+    }
+    icon.innerHTML = `<i class="${iconClassName}" id="${iconId}"></i>`
```

Comment:
prefer the usage of the DOM api to create & append child elements rather than using the `innerHTML` property

Reply:
The problem here is that when i was trying to use classList.add() it threw an error saying that I cant use white space inside class name. The class name is 'fas fa-trash' or 'fas fa-edit' etc. 
This is why i used it. I guess i cant create a map that will map a name I'll create with this name. 
What do you think?

Reply:
`classList.add` may also accept multiple strings at a time (you can call it several times as well but this would be the shorter version), so you'll have something like:
`classList.add('fas', 'fas-trash')`

Reply:
you are right. I changed it and just used icon.className = className.
So this is resolved already